cloud/passportauth/src/auth/PassportAuth.ts

```diff
@@ -0,0 +1,45 @@
+import * as express from 'express';
+import {UserSec} from '../user/UserSec';
+
+export interface Auth {
+  protect(role?: string): void;
+}
+
+export class PassportAuth implements Auth {
+  protected userSec: UserSec;
+  protected loginRoute: string;
+
+  constructor(protected readonly UserSecService: UserSec, loginRoute?: string) {
+    this.userSec = UserSecService;
+    this.loginRoute = loginRoute || '/login';
+  }
+
+  /**
+   * Function which checks if the user requesting access to the resource is authenticated and authorized to
+   * access the resource. Redirects to the login page if user is not authenticated or returns an 'Unauthorized'
+   * response if user does not have the required role.
+   *
+   * @param role {string} [Optional] Role which the user needs in order to access this resource
+   */
+  public protect(role?: string) {
+    return (req: express.Request, res: express.Response, next: express.NextFunction) => {
+      if (req.isAuthenticated()) {
+        if (!role) {
+          return next();
```

Comment:
could add a `console.warn` here when skips happen, very hard to track otherwise

it also serves as a todo for when we include hte logger module